client/server.go

```diff
@@ -58,11 +58,11 @@ func StartHTTPServer(config Config, errChan chan error) {
 	err := consulclient.ConsulInit(consulclient.ConsulConfig{
 		ServiceName:    applicationName,
 		ServicePort:    cfg.Port,
-		ServiceAddress: "localhost",
-		CheckAddress:   "http://localhost:48071/api/v1/ping",
+		ServiceAddress: cfg.Hostname,
+		CheckAddress:   "http://" + cfg.Hostname + ":48071/api/v1/ping",
 		CheckInterval:  "10s",
-		ConsulAddress:  "localhost",
-		ConsulPort:     8500,
+		ConsulAddress:  cfg.ConsulHost,
+		ConsulPort:     cfg.ConsulPort,
```

Comment:
Please remove all Consul stuff from `client/server.go` and move it to the `main.go`. Service should register to Consul in the `main` function, upon init procedure (when connecting to Mongo for example), and not here, just before running the HTTP server.

Reply:
@feclare resons for putting all init code in the main are following:
- connections to various databases (and Consul can be regarded as a type of database) are established on program init - and not at some arbitrary moments in the program when one function from some package is called, as your code currently suggests.
- closing of these functions is deferred to program exit. Otherwise - when you will close a connection to database? Moreover - here, since `http.ListenAndServe` is called in a goroutine we are not blocking and defer would close a databse connection - if this was some DB instead of consul. Now Cosul has a HTTP client that is stateless, but you get the point why conection'ed protocols are put into the init
- finally - since you put all these connections and intis into main, you want - for the sake of redability and maintanability of the program - to find them at the same place. How can I know if the service ever registers to Consul if I look at main - where I expect to see what happens in the program, more or less? I can not. If everybody starts putting funtions in arbitrary places in the code, it will be very hard to maintain it.

To be more precise, even this call to `http.ListenAndServe` should be moved into the main. Package should not contain code of starting the server (especially when this is a standard library function) - this is a duty of the main program - but only describe what this server actually does.

Reply:
I don't agree with them:
* What happens if the connection is closed and we want to reconnect? Do we also need the connections to other microservices initialized in main? We need that logic in the package, not in main.
* I agree now they are closed at exit, we should need a Close function in the package. 

Anyway, I prefer it as is, but if the consensus is having the connections in main I will make it that way.

Reply:
- Connection object is passed from main to package though a function, as we do it now for MongoDB (ex https://github.com/edgexfoundry/export-go/blob/master/cmd/client/main.go#L66)
- Connection to Consul is not a responsibility of package. your package is a library - it should not even have to know that there is Consul. Why should it care? I want to use your library in some other program that does not use Consul - how do I do? Library has to "do one thing and do it well", not to act as a program. This is responsibility of `main`. Do not bloat your package with logic that it is not responsible for.

Reply:
IMHO, `main.go` should server as an integration point, i.e. a place to wire all the dependencies. Therefore, it makes perfect sense to be as explicit as possible and keep it there.